### DISCUSSION TITLE
Why use Github? I am new and have no idea!

### DISCUSSION THREAD

**Discussion by mohamzamir** [community/community#118853] [New to GitHub] [answered]
I am new to Github can someone tell me how to add commit and push? and what is use of all of these?

**Comment by ritessshhh** [comment, marked as answer]
# Working with GitHub

GitHub is essential for version control and collaboration. Here’s how you can add, commit, and push changes to your repository:

## Git Commands

### 1. **Add**
- **Purpose**: Prepares changes for the next commit by adding them to the staging area.
- **Command**:
  - To add a specific file:
    ```bash
    git add <file-name>
    ```
  - To add all modified files:
    ```bash
    git add .
    ```

### 2. **Commit**
- **Purpose**: Saves the staged changes to the local repository. Each commit has an associated message, which is a description explaining why a particular change was made.
- **Command**:
  ```bash
  git commit -m "Your commit message"
  git push origin <branch-name>
